util/patch/patch.go

```diff
@@ -80,11 +80,16 @@ func NewHelper(obj runtime.Object, crClient client.Client, opts ...Option) (*Hel
 }
 
 // Patch will attempt to patch the given object, including its status.
-func (h *Helper) Patch(ctx context.Context, obj runtime.Object) error {
+func (h *Helper) Patch(ctx context.Context, obj runtime.Object, opts ...Option) error {
 	if obj == nil {
 		return errors.Errorf("expected non-nil object")
 	}
 
+	// Calculate the options.
+	for _, opt := range opts {
+		opt.ApplyToHelper(h.opts)
+	}
```

Comment:
These options appear to also apply to any subsequent calls to Patch()? Should it? Do we actually care?

I don't think we expect anyone to actually call Patch() more than once when using the helper and I don't think it would have the expected outcome a user would expect if they do, but we don't prevent them from doing so. 

Should I create an issue to disallow users from calling Patch() multiple times?

Alternatively, I could create an issue for us to support the ability to call Patch() multiple times with the correct behavior.

Reply:
Makes sense, went the other way to allow options to the limited through the scope of a Patch call, ptal.

The change is fine and non-breaking given that we introduce patch helper in v0.3.7